lib/internal/url.js

```diff
@@ -1406,12 +1409,24 @@ function getPathFromURL(path) {
 // as this is the only character that won't be percent encoded by
 // default URL percent encoding when pathname is set.
 const percentRegEx = /%/g;
-function getURLFromFilePath(filepath) {
-  const tmp = new URL('file://');
-  if (filepath.includes('%'))
-    filepath = filepath.replace(percentRegEx, '%25');
-  tmp.pathname = filepath;
-  return tmp;
+function pathToFileURL(filepath) {
+  let resolved = path.resolve(filepath);
+  // path.resolve strips trailing slashes so we must add them back
+  if (filepath.charCodeAt(filepath.length - 1) === CHAR_FORWARD_SLASH &&
+      resolved.charCodeAt(resolved.length - 1) !== CHAR_FORWARD_SLASH)
+    resolved += '/';
```

Comment:
Do we have a test for the trailing slash behavior of `pathToFileURL()`?

Reply:
No explicit test, but coverage through the fact that more than half of the modules tests rely on trailing slash preservation.

Reply:
Could we add an explicit test for trailing forward and back slashes.

Reply:
Hmm, what if it has multiple trailing slashes? For example `/a/////`?

Reply:
The use case here is that it is specifically important to be able to write `new URL('./file', pathToFileURL('dir/'))` and get directory-relative-resolution. The alternative could be to refactor the module loader to not rely on this trailing slash behaviour as well.